internal/ingress/controller/controller.go

```diff
@@ -995,7 +997,15 @@ func (n *NGINXController) createUpstreams(data []*ingress.Ingress, du *ingress.B
 					endp, err := n.serviceEndpoints(svcKey, port.String())
 					if err != nil {
 						klog.Warningf("Error obtaining Endpoints for Service %q: %v", svcKey, err)
+						n.metricCollector.IncOrphanIngress(ing.Namespace, ing.Name, orphanMetricLabelNoService)
 						continue
+					} else {
```

Comment:
I guess you don't need the else here. as you already have the "continue" word above, you will only hit this part of the code on the "else" anyway.

Reply:
Yep, my inadvertence:)))